src/v/kafka/protocol/request_reader.h

```diff
@@ -124,8 +197,11 @@ class request_reader {
     })
     // clang-format on
     std::vector<T> do_read_array(int32_t len, ElementParser&& parser) {
+        if (len < 0) {
+            throw std::out_of_range("Attempt to parse array w/ negative len");
+        }
         std::vector<T> res;
-        res.reserve(std::max(0, len));
```

Comment:
that `res.reserve(std::max(0, len));` line existed makes me still wonder if there was some case where a <0 length was read from a client when decoding a a non-nullable array with the expectation that it be treated as a zero length vector.

Reply:
I think since upstream throws in the negative length scenario we can be comfortable with this change.